src/main/java/org/openrewrite/java/migrate/jakarta/RemoveMethods.java

```diff
@@ -0,0 +1,85 @@
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.jakarta;
+
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.jetbrains.annotations.NotNull;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.NonNull;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.MethodCall;
+
+
+
+public class RemoveMethods extends Recipe {
+
+    @Option(displayName = "Method Pattern",
+      description = "A method pattern for matching required method definition.",
+      example = "*..* hello(..)")
+    @NonNull
+    String methodPattern;
+
+    @Override
+    public @NotNull String getDisplayName() {
+        return "Remove methods calls";
+    }
+
+    @Override
+    public @NotNull String getDescription() {
+        return "Checks for a method patterns and removes the method call from the class";
+    }
+    // All recipes must be serializable. This is verified by RewriteTest.rewriteRun() in your tests.
+    @JsonCreator
+    public RemoveMethods(@NonNull @JsonProperty("methodPattern") String methodPattern) {
+        this.methodPattern = methodPattern;
+    }
+    @Override
+    public @NotNull TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new MethodInvocationVisitor();
+    }
+
+    private class MethodInvocationVisitor extends JavaVisitor<ExecutionContext> {
+        private final MethodMatcher METHOD_PATTERN = new MethodMatcher(methodPattern, false);
+        
+        @Nullable
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ec) {
+            return visitMethodCall(method);
+        }
+
+        @Nullable
+        private <M extends MethodCall> M visitMethodCall(M methodCall) {
+
+            if (!METHOD_PATTERN.matches(methodCall)) {
+                return methodCall;
+            }
+            J.Block parentBlock = getCursor().firstEnclosing(J.Block.class);
+            if (parentBlock != null && !parentBlock.getStatements().contains(methodCall)) {
+                return methodCall;
+            }
+            return null;
+        }
```

Comment:
What's the purpose of this private visitMethodCall instead of having the code directly on visitMethodInvocation?

Reply:
Maybe so it can be applied to `NewClass` instances, too?

Reply:
Hi @joanvr & @knutwannheden I based it on a pattern of code I had found in the repo in another recipe and felt it was better suited for this recipe. I didn't have issues running it directly from visitMethodInvocation as well. So either works I suppose for this recipe.